### DISCUSSION TITLE
Erro IDENTITY_INSERT

### DISCUSSION THREAD

**Discussion by erickcred** [balta-io/forum#54] [Tira Dúvidas] [answered]
Boa tarde, como estão, estou seguindo o processo mas estou com esse erro ao criar os Posts, realizei alguma pesquisas mas não consgui resolver o problema 

Unhandled exception. Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): Não é possível inserir um valor explícito para a coluna de identidade na tabela 'User' quando IDENTITY_INSERT está definido como OFF.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @erickcred, como vai?
A mensagem diz que você tentou enviar um registro sem ID (ou com ID = 0) para o banco, mas o IDENTITY_INSERT, funcionalidade que gera os IDs de forma automática no banco está desabilitado.

Precisa setar o IDENTITY_SEED na tabela, como mostrado no curso de SQL, ou marcar o Id como Key no mapeamento no EF.

**Reply by erickcred** [reply]
consegui identificar o erro estava criando o processo setando assim para pegar os dados
var user = bdc.Users.AsNoTracking().FirstOrDefault(x => x.Id == 8);

mas o correto seria 
var user = bdc.Users.FirstOrDefault(x => x.Id == 1);

para poder fazer essa parte

var post = new Post {
            Author = user,


Muito obrigado a todos pela ajuda
